### PR TITLE
Secure Containers are now emaggable

### PR DESCRIPTION
**Pull request by Robustin**
🆑 Robustin
add: Emags will now permanently disable the locks on digitally locked safes and briefcases. 
/🆑

MFW I have to open code to figure out that only multitool + RNG can currently force open a locked safe/briefcase.

These are digital locks, its completely unintuitive that multitools can scramble the lock but not the device dedicated to scrambling digital security measures. 

After a ~5 second channel the lock will be permanently shorted out and a new animated icon will make it obvious that the storage container is no longer functional.

### CODE REVIEW

File: code/game/objects/items/storage/secure.dm [hunk 1]
```diff
@@ -61,6 +62,20 @@
 	// -> storage/attackby() what with handle insertion, etc
 	return ..()
 
+/obj/item/storage/secure/emag_act(mob/user)
+	user.show_message("<span class='danger'>Now attempting to scramble internal memory.</span>", 1)
```
Comment (Xhuis) [inline code/game/objects/items/storage/secure.dm]: `to_chat(user, "blah blah blah etc.")`

File: code/game/objects/items/storage/secure.dm [hunk 2]
```diff
@@ -61,6 +62,20 @@
 	// -> storage/attackby() what with handle insertion, etc
 	return ..()
 
+/obj/item/storage/secure/emag_act(mob/user)
+	user.show_message("<span class='danger'>Now attempting to scramble internal memory.</span>", 1)
+	src.l_hacking = TRUE
```
Comment (Xhuis) [inline code/game/objects/items/storage/secure.dm]: Remove `src.` from everything with it

File: code/game/objects/items/storage/secure.dm [hunk 3]
```diff
@@ -61,6 +62,20 @@
 	// -> storage/attackby() what with handle insertion, etc
 	return ..()
 
+/obj/item/storage/secure/emag_act(mob/user)
+	user.show_message("<span class='danger'>Now attempting to scramble internal memory.</span>", 1)
+	src.l_hacking = TRUE
+	if (do_after(usr, 50, target = user) && !emagged)
```
Comment (Xhuis) [inline code/game/objects/items/storage/secure.dm]: `user, 50, target = user`

File: code/game/objects/items/storage/secure.dm (hunk 1 above)
Comment (Robustin) [reply code/game/objects/items/storage/secure.dm]: Remind me, how do I designate the span class for to_chat?

**Comment by Hatfish** [comment]
Remove the ability to open these with multitools if you're going to make them emaggable please, there's no point in having a secure briefcase or safe if any chucklefuck can open them.

File: code/game/objects/items/storage/secure.dm (hunk 1 above)
Comment (Xhuis) [reply code/game/objects/items/storage/secure.dm]: The second arg is just a normal string, so you'd do it like:
`to_chat(user, "<span class='notice'>blah bluh</span>")`

File: code/game/objects/items/storage/secure.dm (hunk 2 above)
Comment (Robustin) [reply code/game/objects/items/storage/secure.dm]: Yea, I copied code from the multitool hack, changed both now.

**Comment by Robustin** [comment]
Eh it takes enough time/RNG for the multitool method and I'm sure it would be just as annoying if someone entered a random code and rendered them unusable for the rest of the round if there aren't emags available.

**Comment by RemieRichards** [comment]
No more emag functionality.
@KorPhaeron

**Comment by Robustin** [comment]
I don't support more emag functionality but this is something that belonged here in the first place.

**Comment by RemieRichards** [comment]
no.
we've previously had people PR new emag functionality and we've MADE them move it to the multitool, so this code is already in the preferred form.

the emag is a blight.

File: code/game/objects/items/storage/secure.dm [hunk 4]
```diff
@@ -38,21 +39,21 @@
 				user.show_message("<span class='notice'>You [open ? "open" : "close"] the service panel.</span>", 1)
 			return
 		if ((istype(W, /obj/item/device/multitool)) && (src.open == 1)&& (!src.l_hacking))
-			user.show_message("<span class='danger'>Now attempting to reset internal memory, please hold.</span>", 1)
-			src.l_hacking = 1
-			if (do_after(usr, 100*W.toolspeed, target = src))
+			to_chat(user, text("<span class='danger'>Now attempting to reset internal memory, please hold.</span>")
```
Comment (Xhuis) [inline code/game/objects/items/storage/secure.dm]: Blargh, no reason to use the `text()` wrapper here either! Just `user, "blah"` is fine.

**Comment by Robustin** [comment]
@RemieRichards 

Doesn't change the fact that it always should've been this way from the get-go.

I'm all in favor of removing something dumb from emags but nobody wins when emags do crazy shit like summon holocarp but they can't disable digital locks. Consistency is important.

**Comment by RemieRichards** [comment]
Then be the solution.
Remove the emag.

Be a hero.

**Comment by KorPhaeron** [comment]
When are we going to split the emag into an item that breaks locks and an item that does insane shit like reprogram borgs and summon holocarp

**Comment by DoctoreMad** [comment]
Okay so you're going to stop the emag form being feature creeped to hell... By stopping PRs which are perfectly in-line with intended emag functionality?

**Comment by RemieRichards** [comment]
The intended emag functionality is for there to not be an emag.

**Comment by CPTANT** [comment]
To be honest a lot of emag functionality should be moved to hacking actions.

Removing access however is exactly what the emag is supposed to do.

**Comment by BeeSting12** [comment]
I remember back when the PR making the multitool thing was made. The argument was that there's no card reader on them so an emag wouldn't work.

**Comment by Arianya** [comment]
Theres no card slot on the safe/briefcase to swipe the emag through.

Just give an inspect on the safe/briefcase that hints at the multitool, like we've done for construction.

**Comment by CPTANT** [comment]
"Theres no card slot on the safe/briefcase to swipe the emag through."

Seriously who cares about clumsy justifications like that.

**Comment by nicbn** [comment]
If you can unlock with multitool, unlocking it with emag is not overpowered, it's just making it more intuitive.

**Comment by optimumtact** [comment]
split the emag up into at least two tools, it's time

**Comment by PKPenguin321** [comment]
Is the emag the most bike shedded item in the game? Maintainers shit themselves at the very mention of it

**Comment by PKPenguin321** [comment]
Nobody even uses secure briefcases and a multi tool is easier to get than an emag anyways which only makes this PR even more confusing on both sides

**Comment by RemieRichards** [comment]
No, because bike shedding is about trivial things.
The emag is a cancerous crutch that traitors are forced to rely on, it's more than trivial in the scheme of SS13.

But you know, you don't care about that, you're just using buzzwords because you need that crutch.

**Comment by PKPenguin321** [comment]
Yes I sure was using buzzwords as a crutch in my second comment there

**Comment by RemieRichards** [comment]
Github doesn't always show all comments, especially when viewing from an email.

**Comment by PKPenguin321** [comment]
And yes this is incredibly minor and can fall under bike shedding because when even do people use secure briefcases and in the same round have emags more readily available than multi tools

Oh but sorry I guess I'm just using buzz words

**Comment by RemieRichards** [comment]
You claimed the emag was the most bikeshedded item, but when proven wrong (because you don't understand what bike shedding means) you now say you meant "secure briefcases and emags together"

You are using buzzwords, you are misusing them, and you are a moron.

**Comment by PKPenguin321** [comment]
Your point relating to this pr being

**Comment by RemieRichards** [comment]
That this PR is bad because the emag needs to go, you'd know that if you actually read my comments.
But noooooooooooooooo~ you've got to try and be the big man "Ho ho we don't need to remove this blatant crutch in design, first of all I need that item, and secondly this discussion is pointless! let's ignore the fact I need the item and am just trying to end the discussion!"

**Comment by PKPenguin321** [comment]
Stop using crutch in your arguments its a buzzword
And wtf did I kill you with an emag or something? When did this become personal
This pr is pretty much pointless and whether or not its merged doesnt matter is kind of what my endgame is here

**Comment by RemieRichards** [comment]
If that were true you really wouldn't be so opposed to the idea of the removal of the emag that you feel the need to say that the maintainers "shit themselves" every time the discussion of the removal of an item that doesn't deserve to exist is brought up.

You can try to claim you just think the pr is pointless but your initial words belie that.

**Comment by PKPenguin321** [comment]
>maintainers "shit themselves" every time the discussion of the [emag] comes up

Point in case yeah
Anyways I've said what I wanted to

**Comment by RemieRichards** [comment]
And proved nothing to no-one in the process meaning the only thing pointless here was your own misguided commentary.

**Comment by Robustin** [comment]
This is bad gameplay plain and simple.

We've got a PR right now that has R_WALLS explicitly tell you which tool to deconstruct them - most of us seem to be on board with making gameplay more accessible so you don't have to have a wiki open or 1000 hours of gameplay to understand how the game works. 

Yet here we are bickering about a change with barely any balance implications - regarding an aspect of gameplay that had an old shit like me completely blanked at an answer when the emag wouldn't work. Hell even once I had a multitool I had to open the code and double check because I got a failure message 5 times before it worked.

I even threw in some new icons because I didn't want to make it a lazy bandaid fix. 

PKP is right, this PR should be sitting as Example A in the dictionary for bikeshedding.

**Comment by DoctoreMad** [comment]
RemieRichards when you act like you've shat yourself you should expect people to talk as if you shat yourself
This isn't a formal argument so we aren't pressured to listen to you if you act like you're the boss and if they don't listen to you you'll fire them.

**Comment by RemieRichards** [comment]
>"not liking the thing I don't like is shitting yourself"

You're deluded, your name fits you well.

**Comment by PKPenguin321** [comment]
I certainly implied it

**Comment by Robustin** [comment]
This PR is basically "Emag: The Endless Debate", it's what eclipsed an otherwise complete benign and sane change.

**Comment by Arianya** [comment]
Regardless of the emag debate, I dispute this being a "benign or sane" change, considering the briefcase/safe are meant to be additional security options, however under utilized.

If you want to make a benign change just give it examine text like I suggested *4 days ago*, it'll probably get merged too since it doesn't butt up against the emag's issues.

**Comment by Robustin** [comment]
This isn't worth fighting for, I'll just do the lazy bandaid approach.

**Pull request closed without merging**